### PR TITLE
Axis improvements

### PR DESCRIPTION
**Pull request by ImogenF**
https://visual-tests-axis-improvements.fe.ctmo.io

- Axis labels can be rotated.
- Axis ticks now distinct from rules. 
- Quant axes: tick, label and rule intervals are now configurable.
- Time and categorical axes: rules are now *between* ticks rather than on ticks.
- Axis titles can be added, and font size specified.

### CODE REVIEW

File: packages/visual-tests/src/TestCases/Chart/axes-5.ts [hunk 1]
```diff
@@ -39,12 +39,15 @@ const data: any = {
...
       interval: "day",
+      title: "Axis title",
     },
     y1: {
       type: "quant",
+      title: "Axis title",
     },
     y2: {
       type: "quant",
+      title: "Axis title",
```
Comment (fabien0102) [inline packages/visual-tests/src/TestCases/Chart/axes-5.ts]: Can you change the title here, so we can see the difference in the visual resul

File: packages/visualizations/src/Chart/axes/axis_utils.ts [hunk 2]
```diff
@@ -73,13 +76,10 @@ export const alignAxes = (axes: { [key: string]: AxisClass<any> }) => {
...
-  const computed: { [key: string]: Partial<AxisComputed> } = mapValues(
-    (axis: AxisClass<number>): Partial<AxisComputed> => axis.computeInitial(),
-  )(axes)
-
-  const axisKeys: string[] = keys(computed) as string[]
-  const intervalOne: any = axes[axisKeys[0]].interval
-  const intervalTwo: any = axes[axisKeys[1]].interval
+  const computed = mapValues((axis: AxisClass<number>) => axis.computeInitial())(axes)
+  const axisKeys = keys(computed) as string[]
+  const intervalOne = axes[axisKeys[0]].options.interval
```
Comment (fabien0102) [inline packages/visualizations/src/Chart/axes/axis_utils.ts]: You have a type error somewhere here 😉 

From netlify:
```bash
11:24:12 AM: lerna ERR! src/Chart/axes/axis_utils.ts(81,49): error TS2339: Property 'interval' does not exist on type 'AxisOptions'.
11:24:12 AM: lerna ERR!   Property 'interval' does not exist on type 'CategoricalAxisOptions'.
11:24:12 AM: lerna ERR! src/Chart/axes/axis_utils.ts(82,49): error TS2339: Property 'interval' does not exist on type 'AxisOptions'.
11:24:12 AM: lerna ERR!   Property 'interval' does not exist on type 'CategoricalAxisOptions'.
```

**Comment by ImogenF** [comment]
@fabien0102 should be fixed now.

**Comment by fabien0102** [comment]
@ImogenF Nice, I'm just waiting to have #485 merge to rebase your branch, and hopefully have everything green on Travis 🤞

**Comment by TejasQ** [comment]
[Charts](https://trello.com/c/e80n3s1K/151-charts)

**Comment by TejasQ** [comment]
Soooo lots of approving responses here. Can we merge this?

**Comment by ImogenF** [comment]
@TejasQ See fabien's comment above - waiting for another branch to be merged and for everything to pass in Travis

**Comment by TejasQ** [comment]
I think we ought to rebase this on the master and then everything _should_ magically pass in Travis. ✨

**Comment by ImogenF** [comment]
Will do.

**Pull request merged**